MTGSDKSwift/MagicalCardManager.swift

```diff
@@ -8,74 +8,127 @@
+        }
+        
+        guard let url = URLBuilder.buildURLWithParameters(parameters, andConfig: configuration) else {
+            networkError = NetworkError.miscError("fetchCards url build failed")
+            return
+        }
+        
+        mtgAPIService.mtgAPIQuery(url: url) {
+            result in
+            switch result {
+            case .success(let json):
+                let cards = Parser.parseCards(json: json)
+                completion(Result.success(cards))
+            case .error(let error):
+                networkError = error
+            }
+        }
+    }
     
-    public static var enableLogging: Bool = false
-    public var fetchPageSize: String = "12"
-    public var fetchPageTotal: String = "1"
+    /**
+     Reteives an array of CardSet which matches the parameters given
+     - parameter parameters: [SetSearchParameter]
+     - parameter configuration: MTGSearchConfiguration
+     - parameter completion: (Result<[CardSet]>)-> Void
+     */
     
-    private let mtgAPIService = MTGAPIService()
-    private var urlManager: URLManager {
-        let urlMan = URLManager()
-        urlMan.pageTotal = fetchPageTotal
-        urlMan.pageSize = fetchPageSize
-        return urlMan
+    public func fetchSets(_ parameters: [SetSearchParameter],
+                          _ configuration: MTGSearchConfiguration = MTGSearchConfiguration.defaultConfiguration,
+                          _ completion: @escaping SetCompletion) {
+        var networkError: NetworkError? {
+            didSet {
+                completion(Result.error(networkError!))
+            }
+        }
+        
+        guard let url = URLBuilder.buildURLWithParameters(parameters, andConfig: configuration) else {
+            networkError = NetworkError.miscError("fetchSets url build failed")
+            return
+        }
+        
+        mtgAPIService.mtgAPIQuery(url: url) {
+            result in
+            switch result {
+            case .success(let json):
+                let sets = Parser.parseSets(json: json)
+                completion(Result.success(sets))
+            case .error(let error):
+                networkError = error
+            }
+        }
     }
-    private let parser = Parser()
     
     /**
      Fetch JSON returns the raw json data rather than an Array of Card or CardSet. It will return json for sets or cards depending on what you feed it
-     
-     
      - parameter parameters: either [CardSearchParameter] or [SetSearchParameter]
-      - parameter completion: ([String:Any]?, NetworkError?) -> Void
-     
+     - parameter config: MTGSearchConfiguration
+     - parameter completion: (Result<JSONResults>) -> Void
     */
-    public func fetchJSON(_ parameters: [SearchParameter], completion: @escaping JSONCompletionWithError) {
+    public func fetchJSON(_ parameters: [SearchParameter],
+                          _ configuration: MTGSearchConfiguration = MTGSearchConfiguration.defaultConfiguration,
```

Comment:
1. For brevity, you could rewrite this line to: `configuration: MTGSearchConfiguration = .defaultConfiguration` (note the removal of the `MTGSearchConfiguration` prefix).
2. The standard (conventional) parameter naming convention is to only use the `_` for the first parameter and use names after that.  I'd suggest following that convention for readability.